File: solve/lua/exeter.lua
#! /usr/bin/lua

require "common"

function recurse(k)
	if k == 1 then
		output()
	else
		for i = 1, k do
			swap(i, k)
			recurse(k - 1)
			swap(i, k)
		end
	end
end

skeleton(UNORDERED, function ()
	recurse(n)
end)

File: solve/cpp/exeter.cc
#include "common.hh"

void recurse(int k)
{
    if (k == 0)
    {
        output();
    } 
    else
    { 
        for (int i = 0; i <= k; ++i)
        {
            char temp = a[i];
            a[i] = a[k];
            a[k] = temp;
            recurse(k - 1);
            a[k] = a[i];
            a[i] = temp;
        } 
    }
}

int main(int argc, char *argv[])
{
    if (init(argc, argv, UNORDERED) == 0)
    { 
        recurse(n - 1); 
        return final();
    }
    return 1;
}
